### PR TITLE
Implement json appender for log4js

### PR DESCRIPTION
**Pull request by manoskouvarakis**

### CODE REVIEW

File: src/initializers/log4js/json-appender.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import * as lodash from 'lodash';
+
+const jsonAppender = ({ filter_status = [] }) => {
```
Comment (nikostoulas) [inline src/initializers/log4js/json-appender.ts]: Do we use the filter_status? It is used in honeybadger appender to not send statuses that are expected eg: 401 unauthorized.

File: src/initializers/log4js/json-appender.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  return logEvent => {
+    let event = logEvent.level.levelStr === 'ERROR' ? createErrorLog(logEvent) : createValidLog(logEvent);
+    var json = JSON.stringify(event);
+    console.log(json);
```
Comment (nikostoulas) [inline src/initializers/log4js/json-appender.ts]: What does log4js use internally? Maybe process.stdout.write is more suitable

File: src/initializers/log4js/json-appender.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    message: logEvent.data[0].message,
+    stack_trace: logEvent.data[0].stack
+  };
+};
+
+const createValidLog = logEvent => {
+  const data = lodash.flattenDeep(logEvent.data);
+  const message = data
+    .map(element => {
+      return typeof element === 'object' ? JSON.stringify(element) : element;
```
Comment (nikostoulas) [inline src/initializers/log4js/json-appender.ts]: Can we use append the object keys to the returned obj?

File: src/initializers/log4js/json-appender.ts (hunk 1 above)
Comment (manoskouvarakis) [reply src/initializers/log4js/json-appender.ts]: Done

File: src/initializers/log4js/json-appender.ts (hunk 2 above)
Comment (manoskouvarakis) [reply src/initializers/log4js/json-appender.ts]: Done (kept console as it has only memory overhead)

File: src/initializers/log4js/json-appender.ts (hunk 3 above)
Comment (manoskouvarakis) [reply src/initializers/log4js/json-appender.ts]: Done

File: src/initializers/log4js/json-appender.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  };
+};
+
+const createValidLog = logEvent => {
+  const data = lodash.flattenDeep(logEvent.data);
+  let context = {};
+  const message = data
+    .map(element => {
+      if (typeof element === 'object') {
+        Object.assign(context, element);
```
Comment (nikostoulas) [inline src/initializers/log4js/json-appender.ts]: It's a bit weird that a map has side effects. Can we refactor it to use forEach instead?

File: src/initializers/log4js/json-appender.ts (hunk 2 above)
Comment (nikostoulas) [reply src/initializers/log4js/json-appender.ts]: 👍

File: src/initializers/diamorphosis.ts [hunk 5]
```diff
@@ -5,6 +5,7 @@ export default (config, orkaOptions: Partial<OrkaOptions>) => {
   config.nodeEnv = config.nodeEnv || 'development';
   config.honeybadgerApiKey = config.honeybadgerApiKey || '';
   config.honeybadgerEnvironment = config.honeybadgerEnvironment || '';
+  config.honeybadgerFilterStatus = config.honeybadgerFilterStatus || [];
```
Comment (nikostoulas) [inline src/initializers/diamorphosis.ts]: 👍

File: src/initializers/log4js/json-appender.ts (hunk 4 above)
Comment (manoskouvarakis) [reply src/initializers/log4js/json-appender.ts]: Done

File: src/initializers/diamorphosis.ts [hunk 6]
```diff
@@ -23,5 +30,4 @@ export default (config, orkaOptions: Partial<OrkaOptions>) => {
     ...config.riviere
   };
   diamorphosis(orkaOptions.diamorphosis);
-  config.honeybadgerEnvironment = config.honeybadgerEnvironment || config.nodeEnv;
```
Comment (nikostoulas) [inline src/initializers/diamorphosis.ts]: Can you add this here again? And possibly add a test for it.
The problem is that if you don't set honeybadgerEnvironment but only set NODE_EVN in env the honeybadgerEnvironment remains 'development'

File: src/initializers/diamorphosis.ts [hunk 7]
```diff
@@ -1,14 +1,21 @@
...
+    apiKey: '',
+    environment: config.nodeEnv,
+    filterStatus: '',
+    ...config.honeybadger
+  };
+  config.printLogo = defaultTo(config.printLogo, true);
   config.log = {
     pattern: '%[[%d] [%p] %c%] %m',
     level: 'debug',
+    consoleLogging: true,
```
Comment (nikostoulas) [inline src/initializers/diamorphosis.ts]: LOG_CONSOLE, LOG_JSON seems better wdyt?

(So log: {console: true, json:false})

**Pull request merged**